src/applications/gi/components/profile/CalculatorForm.jsx

```diff
@@ -31,6 +32,12 @@ class CalculatorForm extends React.Component {
     this.props.onInputChange({ field, value });
   }
 
+  handleBeneficiaryZIPCodeChanged = (event) => {
+    if (!event.dirty) {
```

Comment:
I don't understand why this input component is firing an onchange event onblur but this dirty flag seems to be the way to differentiate between the two events.

Reply:
Those Errorable components treat the "value" as an object containing the input value and a dirty flag, so it calls onValueChange when either of those changes.

I would really like to get some standalone input components built based on us-forms-system, but for now we have these quirky ones.